### DISCUSSION TITLE
I don't get segmentation instance masks with yolov8 using SAHI

### DISCUSSION THREAD

**Discussion by ThibautMrl** [obss/sahi#1000] [Q&A] [answered]
Hi everyone, I followed the examples available on the SAHI github and tried to help myself with other examples from the internet but I still can't get the segment masks when I use yolov8.
Here's my code:
```python
from sahi.utils.yolov8 import download_yolov8s_model
from sahi import AutoDetectionModel
from sahi.utils.cv import read_image
from sahi.utils.file import download_from_url
from sahi.predict import get_prediction, get_sliced_prediction, predict
from pathlib import Path
from IPython.display import Image
# Download YOLOv8 model
yolov8_model_path = "models/yolov8s-seg.pt"
download_yolov8s_model(yolov8_model_path)
# Download test images
download_from_url('https://raw.githubusercontent.com/obss/sahi/main/demo/demo_data/small-vehicles1.jpeg', 'demo_data/small-vehicles1.jpeg')
download_from_url('https://raw.githubusercontent.com/obss/sahi/main/demo/demo_data/terrain2.png', 'demo_data/terrain2.png')
detection_model = AutoDetectionModel.from_pretrained(
    model_type='yolov8',
    model_path=yolov8_model_path,
    confidence_threshold=0.3,
    device="cpu",  # or 'cuda:0'
)
result = get_sliced_prediction(
    "demo_data/small-vehicles1.jpeg",
    detection_model,
    slice_height=256,
    slice_width=256,
    overlap_height_ratio=0.2,
    overlap_width_ratio=0.2
)
print(result.object_prediction_list)
result.export_visuals(export_dir="demo_data/")
Image("demo_data/prediction_visual.png")
```
And here's a part of the result but there is always `mask : None` :
```
[ObjectPrediction<
    bbox: BoundingBox: <(447.6741943359375, 309.5724792480469, 495.7220458984375, 342.0692138671875), w: 48.0478515625, h: 32.496734619140625>,
    mask: None,
    score: PredictionScore: <value: 0.8857606649398804>,
    category: Category: <id: 2, name: car>>, ObjectPrediction<
    bbox: BoundingBox: <(321.2298583984375, 322.0831298828125, 382.9784851074219, 363.1929473876953), w: 61.748626708984375, h: 41.10981750488281>,
    mask: None,
    score: PredictionScore: <value: 0.8725772500038147>,
    category: Category: <id: 2, name: car>>, ObjectPrediction<
    bbox: BoundingBox: <(832.658935546875, 308.4786148071289, 874.089599609375, 344.8790588378906), w: 41.4306640625, h: 36.40044403076172>,
    mask: None,
    score: PredictionScore: <value: 0.8661474585533142>,
    category: Category: <id: 2, name: car>>
```

Can anyone tell me what's wrong with my code?

**Comment by kyra-smith** [comment]
I'm having the same issue and would love to know if there is a solution for this!

**Reply by fcakyon** [reply]
its now live on the latest version @kyra-smith ! https://github.com/obss/sahi/discussions/1051

check the demo notebook on how it works: https://github.com/obss/sahi/blob/main/demo/inference_for_yolov8.ipynb

**Comment by jmarintur** [comment]
Hi @ThibautMrl and @kyra-smith, as you can see, `.has_mask` returns `False`. 

![image](https://github.com/obss/sahi/assets/28574023/07b13d0f-7e6c-4f2c-920e-6cc396c6acb9)

Which means that this model does not return masks. If you dig a little bit into the code:

https://github.com/obss/sahi/blob/d91e1e6a1e62f966540abe7471f88bf792de0f93/sahi/models/yolov8.py#L87-L91

I am afraid that yolov8 models in sahi do not return masks.

**Reply by ThibautMrl** [reply]
Hello, thank you for your reply. It's a shame that segmentation isn't implemented for yolov8.

**Reply by fcakyon** [reply]
it now live on the latest version @jmarintur @ThibautMrl ! https://github.com/obss/sahi/discussions/1051

check the demo notebook on how it works: https://github.com/obss/sahi/blob/main/demo/inference_for_yolov8.ipynb

**Comment by navtuteja** [comment]
I am also looking to get the masks from the results. Any way to get it?

**Reply by fcakyon** [reply]
its now live on the latest version @navtuteja ! https://github.com/obss/sahi/discussions/1051

check the demo notebook on how it works: https://github.com/obss/sahi/blob/main/demo/inference_for_yolov8.ipynb

**Comment by fcakyon** [comment, marked as answer]
Segmentation mask support for yolov8 and mmdet models is live with the latest release! https://github.com/obss/sahi/discussions/1051

Check the demo notebook on how it works: https://github.com/obss/sahi/blob/main/demo/inference_for_yolov8.ipynb

**Comment by MikeBai426** [comment]
I figured it out: I need an updated SAHI version to run the code. 
pip install -U ultralytics sahi
